jwst_kpi/trim_frames/trim_frames_step.py

```diff
@@ -0,0 +1,194 @@
+import os
+
+import matplotlib.pyplot as plt
+import numpy as np
+from jwst import datamodels
+from jwst.stpipe import Step
+from scipy.ndimage import median_filter
+
+from .. import utils as ut
+from ..datamodels import TrimmedCubeModel, TrimmedImageModel
+from .trim_frames_plots import plot_trim
+
+
+class TrimFramesStep(Step):
+    """
+    Trim frames.
+
+    Parameters
+    -----------
+    input_data :  ~jwst_kpi.datamodels.KPFitsModel
+        Single filename for extracted kernel phase data
+    plot : bool
+        Generate plots
+    show_plots : bool
+        Show plots
+    previous_suffix : Optional[str]
+        Suffix of previous file. DEPRECATED: use ~input_data directly instead
+    good_frames : List[int]
+        List of good frames, bad frames will be skipped.
+    trim_cent : List[int]
+        Center around which to trim (X and Y coordinates, list of length 2)
```

Comment:
I don't know about the other instruments, but for MIRI the coordinates are reversed. The list of ints of trim_cent therefore looks like this: [y, x]. If this is also the case for other instruments, perhaps it would be good to specify it?